crates/sui-indexer/src/apis/write_api.rs

```diff
@@ -32,9 +35,10 @@ impl WriteApiServer for WriteApi {
         tx_bytes: Base64,
         signatures: Vec<Base64>,
         request_type: ExecuteTransactionRequestType,
```

Comment:
is it possible to use the option to determine the request type?

Reply:
I think they should be orthogonal, people may want the transaction to be locally executed but do not want the effects

Reply:
if people don't need the effect straight away, they can wait for state sync for the local execution?

Reply:
I agree that this should be the common case - one edge cases I can think of is: A DApp maintains a list of object ids, they wants to re-fetch this list of objects after the user submits a transaction, but they do not want any effects while wanting to get the latest data.

Maybe I can make `request_type` an optional variable and derive it from `options` if None? wdyt?

Reply:
Synced offline:

> Patrick for the new object changes and balance changes data, (separated from event), will only be available after local execution, as they need to access the object data, if user choose WaitForEffectsCert but choose BalanceChanges as well, Balance change will have to override WaitForEffectsCert.... will be a bit strange lol

> Chris: sounds good, I'll remove request_type for now and in the future if needed we can add it back as an optional

Reply:
Synced again

Chris: what I mentioned(do not want effects but want local execution) is actually a common case in many of our tests: https://github.com/MystenLabs/sui/blob/0daaca6e69c49e6132cdcfe089eafd52fb2bb26f/crates/sui-json-rpc/src/unit_tests/rpc_server_tests.rs#L536-L543
I think what I will do is to still keep request_type as an optional parameter that will be derived from options in most cases. In most cases, users will leave request_type as None unless they know what they are doing.

> if the user chooses WaitForEffectsCert but chooses BalanceChanges as well,
The API can check for this and return an error if the user inputs conflicting types